cli/azd/test/mocks/mockgraphsdk/mocks.go

```diff
@@ -15,7 +15,7 @@ import (
 
 func CreateGraphClient(mockContext *mocks.MockContext) (*graphsdk.GraphClient, error) {
 	clientOptions := CreateDefaultClientOptions(mockContext)
-
+	clientOptions.Retry.RetryDelay = -1
```

Comment:
This feels a little unfortunate, since it impacts our retry behavior for the actual product cases and makes us a little less well behaved when talking to service.

I saw this fun little trick in https://github.com/Azure/azure-sdk-for-go/blob/5f013ebffdb3f090b487c7b181880dbc98610902/sdk/azcore/runtime/poller.go#L237-L240 where they vary behavior when running under test.  I wonder if a similar guard here (or perhaps centralized in `CreateDefaultClientOptions`) that would only set RetryDelay to -1 when we are running under test would be nicer?

Reply:
> This feels a little unfortunate, since it impacts our retry behavior for the actual product cases and makes us a little less well behaved when talking to service.

Not following completely yet. Would we expect `mockgraphsdk.CreateGraphClient` to be called in product code?

>  I wonder if a similar guard here (or perhaps centralized in CreateDefaultClientOptions) that would only set RetryDelay to -1 when we are running under test would be nicer?

I'd need to think more about that. I'm a little worried about implications of having product behavior change with test detection logic. Certainly the implications are overblown with something like `test.v`. In the az-sdk-for-go example, they're loosening constraints which is less big of a deal than changing configuration.

Personally, I'd prefer if such enforcement happens as part of test setup. And if we fail to set the right retry behavior, letting tests run longer is fine. Over time we may find a consistent pattern here.

Reply:
> Not following completely yet. Would we expect mockgraphsdk.CreateGraphClient to be called in product code?

I was missing the forest (the fact that was in the mockgraphsdk) for the trees (the actual line in the diff).  I'm not concerned with the change, thanks for pointing the obvious out to me and sorry for the distraction.

Reply:
No worries at all. The graphsdk package is structured quite differently than other packages and easily missed.. And I just brought up the latter discussion because I do think we'll need to do something in the future.